addons/website_sale/models/product_wishlist.py

```diff
@@ -0,0 +1,35 @@
+from datetime import datetime, timedelta
+from odoo import api, fields, models, _
+from odoo.http import request
+
+
+class ProductWishlist(models.Model):
+    _name = 'product.wishlist'
+    _description = 'Product Wishlist'
+    _sql_constraints = [
+        ("product_unique_partner_id",
+         "UNIQUE(product_id, partner_id)",
+         "Duplicated wishlisted product for this partner."),
+    ]
+
+    partner_id = fields.Many2one('res.partner', string='Owner')
+    product_id = fields.Many2one('product.product', string='Product', required=True)
+    currency_id = fields.Many2one('res.currency', related='pricelist_id.currency_id', readonly=True)
+    pricelist_id = fields.Many2one('product.pricelist', string='Pricelist', help='Pricelist when added')
+    price = fields.Monetary(currency_field='currency_id', string='Price',
+                            help='Price of the product when it has been added in the wishlist')
+    website_id = fields.Many2one('website', ondelete='cascade', required=True)
+    active = fields.Boolean(default=True, required=True)
+    stock_notification = fields.Boolean(default=False, required=True)
+
+    @api.model
+    def _add_to_wishlist(self, pricelist_id, currency_id, website_id, price, product_id, partner_id=False):
+        wish = self.env['product.wishlist'].create({
+            'partner_id': partner_id,
+            'product_id': product_id,
+            'currency_id': currency_id,
+            'pricelist_id': pricelist_id,
+            'price': price,
+            'website_id': website_id,
+        })
+        return wish
```

Comment:
Since the route calling this method is public, we also need to check that the wishlist does not exist, we would otherwise trigger the unicity rules defined above.
```suggestion
    def _add_to_wishlist(self, pricelist_id, currency_id, website_id, price, product_id, partner_id=False):
        wish = self.env['product.wishlist'].sudo().search([('partner_id', '=', partner_id), ('product_id', '=', product_id)]) # Unsure about sudo here
        if wish:
            # Maybe update price pricelist etc
        else:
            wish = self.env['product.wishlist'].create({
                'partner_id': partner_id,
                'product_id': product_id,
                'currency_id': currency_id,
                'pricelist_id': pricelist_id,
                'price': price,
                'website_id': website_id,
            })
        return wish
```